### PR TITLE
feat(core): make reference fields searchable in list previews

### LINKED ISSUE #4775: List Previews search ignores reference fields in subtitle section
I'm not sure if this is a bug or a feature so please forgive me if I'm wrong

**Describe the bug**

As the title says, the List Preview search ignores reference fields in the subtitle section of a List item, while works fine with _regular_ fields 

**To Reproduce**

1. Create a schema with a reference field
2. Use that reference field in the subtitle section of a List item
3. Use the _regular_ field in the subtitle section of a List item
4. Search for that reference field -> No results
5. Search for the _regular_ field -> Search result

**Expected behavior**

I believe that the reference fields should not be discriminated by the search functionality

**Which versions of Sanity are you using?**

v3.14.1

**What operating system are you using?**

elementaryOS Jolnir (Ubuntu 20.04.6 LTS based)

**Which versions of Node.js / npm are you running?**

npm: v9.5.1
node: v18.16.0

### PR DESCRIPTION
**Pull request by jordanl17**
### Description

Fields shown in a list preview through a reference - for example `subtitle: 'author.name'` - were not searchable, while regular fields were. A previous fix (#12925) compiled these paths into dereferenced GROQ (`author->name`) and was reverted: under the now-default `groq2024` strategy that crashed all search with "score() function received unexpected expression", and the equivalent filter form forced a non-index-accelerated per-document join. Fixes #4775 (SAPP-3879).

This PR makes reference-preview fields searchable without any query-time dereference, using a two-phase approach: phase one resolves the ids of referenced documents whose preview leaf matches the term via an index-accelerated `_type == … && leaf match …` query; phase two surfaces the referring documents by boosting `references($ids)` inside `score()`. Both `references()` and the resolution query are index-accelerated, so the per-document join that caused the revert does not recur. When a searched type has no reference-traversing preview path, the query is byte-identical to before and phase one is skipped.

### What to review

- `deriveReferenceSearchSpecs.ts` - walks `preview.select`, follows a single reference hop, and derives `{targetType, leafPath, weight}` for string/PT/slug leaves; array paths and chained references are intentionally skipped. The spec cache is keyed per `maxDepth`.
- `createReferenceResolveQuery.ts` - the phase-one query. Note the slice is `[0...1000]._id`, not `._id[0...1000]` (the latter binds the slice to the `_id` string and returns nulls), and the bound is a literal because GROQ rejects a parameterised slice. Matching beyond 1000 ids is best-effort.
- `createSearchQuery.ts` - injects `boost(references($__refIds), weight)` into `score()` and ORs `references($__refIds)` into the unscored filter path; output is unchanged when no reference ids are passed.
- `createGroq2024Search.ts` - orchestrates phase one then phase two, normalising resolved ids through `getPublishedId` since references always point at the published id. If phase one fails, it falls back to a plain search rather than failing the base term-match.

before-after

### Testing

Unit tests cover spec derivation (including chained-reference, slug, PT, multi-target, and `maxDepth`-boundary cases), the phase-one query shape, the score/filter injection, and the two-phase orchestration (sequence, id normalisation, skip-when-no-specs, and the phase-one-failure fallback). Tests assert no `->` ever reaches `score()` or a filter, and parse the phase-one query through `groq-js` to catch syntax regressions. Verified end-to-end against a live dataset: searching an author's name surfaces books that reference that author (titles not containing the term), both phase queries return 200, and the old `->` form was confirmed to still produce the original crash. Measured latency shows `references()` with a large id array is not a performance cliff; the added cost is one index-accelerated round-trip on searches over reference-preview types.

### Notes for release

List preview and global search now match on reference fields used in a document's preview (for example a `subtitle` of `author.name`), not just direct fields.

### CODE REVIEW

File: packages/sanity/src/core/search/groq2024/createGroq2024Search.ts [hunk 1]
```diff
@@ -41,31 +45,77 @@ export const createGroq2024Search: SearchStrategyFactory<Groq2024SearchResults>
...
+        {referenceIds, referenceWeight: maxReferenceWeight},
+      )
 
-        // Search overfetches by 1 to determine whether there is another page to fetch. Therefore,
-        // the penultimate result must be used to determine the start of the next page.
-        const lastResult = hasNextPage ? hits.at(-2) : hits.at(-1)
+      return client.observable.fetch<SanityDocumentLike[]>(query, params, options).pipe(
+        map((hits) => {
+          const hasNextPage =
+            typeof searchOptions.limit !== 'undefined' && hits.length > searchOptions.limit
```
[the remark this answers is not in the document]
Comment (jordanl17) [reply packages/sanity/src/core/search/groq2024/createGroq2024Search.ts:75]: Good catch - fixed in 106d616546. The result now slices by `searchOptions.limit ?? params.__limit - 1`, so the overfetched element is excluded even when no limit is passed (added a test covering the no-limit case). Note this was pre-existing behaviour relocated into `runMainSearch` by this PR, and latent in practice since the list-pane and global search always pass a limit.

**Review by bjoerge — APPROVED** [review]
